.github/workflows/build.yml

```
@@ -260,84 +260,28 @@ jobs:
       free-threading: ${{ matrix.free-threading }}
       os: ${{ matrix.os }}
 
-  build-ubuntu-ssltests-openssl:
-    name: 'Ubuntu SSL tests with OpenSSL'
+  build-ubuntu-ssltests:
+    name: 'Ubuntu SSL tests'
     runs-on: ${{ matrix.os }}
```

All jobs run on the same OS, this is simpler:
```suggestion
    runs-on: ubuntu-24.04
```

---

We use `matrix.os` in the cache key so better not hardcode it here?